src/coprocessor/dag/pipeline.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use kvproto::coprocessor::KeyRange;
+use tipb::executor::{self, ExecType};
+
+use storage::Store;
+
+use super::executor::{
+    Executor, HashAggExecutor, IndexScanExecutor, LimitExecutor, SelectionExecutor,
+    StreamAggExecutor, TableScanExecutor, TopNExecutor,
+};
+use coprocessor::dag::expr::EvalConfig;
+use coprocessor::*;
+
+pub struct ExecutorPipelineBuilder;
+
+impl ExecutorPipelineBuilder {
+    pub fn build_normal<S: Store + 'static>(
```

Comment:
We need some comments for these `pub` functions and traits.

Reply:
+1

Reply:
Thanks a lot for mentioning! I forgot adding it 🤣